src/coreclr/jit/lower.cpp

```diff
@@ -8007,8 +8000,23 @@ void Lowering::LowerStoreIndirCoalescing(GenTreeStoreInd* ind)
             return;
         }
 
-        // Offset has to match the size of the type. We don't support the same or overlapping offsets.
-        if (abs(prevData.offset - currData.offset) != (int)genTypeSize(prevData.targetType))
+        const int offset = abs(prevData.offset - currData.offset);
+
+        // Offset has to match the size of the type.
+        if (offset != (int)genTypeSize(prevData.targetType))
+        {
+            if (offset == 0)
```

Comment:
This is basically doing:

```c++
if (prevData.offset == currData.offset)
{
  continue;
}
```

I didn't understand the comment and need for `if (offset != (int)genTypeSize(prevData.targetType))`.

Reply:
you could have the `if (offset != (int)genTypeSize(prevData.targetType))` condition separately.

if (prevData.offset == currData.offset)
{
  continue;
}
if (offset != (int)genTypeSize(prevData.targetType))
{
  return;
}

Reply:
Ah, you're right - addressed